models/rubicon/optimism/rubicon_optimism_offers.sql

```diff
@@ -0,0 +1,214 @@
+    ((CAST(txn.gas_used AS DECIMAL(38,0)) / power(10, 18)) * CAST(txn.gas_price AS decimal(38,0))) + ((CAST(txn.l1_gas_used AS DECIMAL(38,0)) / power(10, 18)) * CAST(txn.l1_gas_price AS decimal(38,0)) * CAST(txn.l1_fee_scalar AS decimal(38,0))) AS txn_cost_eth,
+    eth.price AS eth_price,
+    ((CAST(txn.gas_used AS DECIMAL(38,0)) / power(10, 18)) * CAST(txn.gas_price AS decimal(38,0))) + ((CAST(txn.l1_gas_used AS DECIMAL(38,0)) / power(10, 18)) * CAST(txn.l1_gas_price AS decimal(38,0)) * CAST(txn.l1_fee_scalar AS decimal(38,0))) * eth.price AS txn_cost_usd,
+    offers.project_contract_address, 
+    offers.tx_hash,
+    txn.from AS tx_from,
+    txn.to AS tx_to
+FROM offers
+
+-- get the transaction data for the offer
+INNER JOIN {{ source('optimism', 'transactions') }} txn
+    ON txn.hash = offers.tx_hash
+    AND txn.block_number = offers.block_number
+    {% if not is_incremental() %}
+    AND txn.block_time >= cast('{{ project_start_date }}' AS timestamp)
+    {% endif %}
+    {% if is_incremental() %} -- only run this filter if it is an incremental run
+    AND txn.block_time >= date_trunc('day', now() - interval '1 week')
+    {% endif %}
+    
+-- get the relevant sell token data
+LEFT JOIN {{ ref('tokens_erc20') }} erc20_sell
+    ON erc20_sell.contract_address = offers.sell_token_address
+    AND erc20_sell.blockchain = 'optimism'
+
+-- get the relevant buy token data
+LEFT JOIN {{ ref('tokens_erc20') }} erc20_buy
+    ON erc20_buy.contract_address = offers.buy_token_address
+    AND erc20_buy.blockchain = 'optimism'
+
+-- get the price data for the sell token
+LEFT JOIN {{ source('prices', 'usd') }} sell_token_price
+    ON sell_token_price.minute = date_trunc('minute', offers.block_time)
+    AND sell_token_price.contract_address = offers.sell_token_address
+    AND sell_token_price.blockchain = 'optimism'
+    {% if not is_incremental() %}
+    AND sell_token_price.minute >= cast('{{ project_start_date }}' AS timestamp)
+    {% endif %}
+    {% if is_incremental() %}
+    AND sell_token_price.minute >= date_trunc('day', now() - interval '1 week')
+    {% endif %}
+    
+-- get the price data for the buy token
+LEFT JOIN {{ source('prices', 'usd') }} buy_token_price
+    ON buy_token_price.minute = date_trunc('minute', offers.block_time)
+    AND buy_token_price.contract_address = offers.buy_token_address
+    AND buy_token_price.blockchain = 'optimism'
+    {% if not is_incremental() %}
+    AND buy_token_price.minute >= cast('{{ project_start_date }}' AS timestamp)
+    {% endif %}
+    {% if is_incremental() %}
+    AND buy_token_price.minute >= date_trunc('day', now() - interval '1 week')
+    {% endif %}
+
+-- get the price of eth at the time of the offer 
+LEFT JOIN {{ source('prices', 'usd') }}  eth
+    ON eth.minute = date_trunc('minute', offers.block_time)
+    AND cast(eth.contract_address AS varchar) = '0xdeaddeaddeaddeaddeaddeaddeaddeaddead0000' -- this is for optimism specifically
```

Comment:
Hi @denverbaumgartner, we may need to fix the cast to varchar here.
```suggestion
    AND cast(eth.contract_address AS varchar(100)) = '0xdeaddeaddeaddeaddeaddeaddeaddeaddead0000' -- this is for optimism specifically
```

Reply:
hey @Hosuke, just fixed this, thanks for the suggestion 💯